python/ray/serve/controller.py

```diff
@@ -487,6 +499,9 @@ async def __init__(self,
         self.actor_reconciler._start_routers_if_needed(
             self.http_host, self.http_port, self.http_middlewares)
 
+        # Map of awaiting results
+        self.inflight_results: Dict[ResultId, asyncio.Future] = dict()
```

Comment:
Should this be an `asyncio.Event`, not an `asyncio.Future` for the final design? Should work here as well.

Reply:
I switched to using a future as per: https://github.com/ray-project/ray/pull/12577#discussion_r534579106